lib/integrations/externalcloudaudit/bootstrap_test.go

```diff
@@ -0,0 +1,226 @@
+	if _, ok := c.buckets[*params.Bucket]; !ok {
+		// bucket doesn't exist return no such bucket error
+		return nil, &s3types.NoSuchBucket{Message: aws.String("The bucket doesn't not exist")}
+	}
+
+	return &s3.PutBucketLifecycleConfigurationOutput{}, nil
+}
+
+func (c *fakeBootstrapInfraClient) CreateWorkGroup(ctx context.Context, params *athena.CreateWorkGroupInput, optFns ...func(*athena.Options)) (*athena.CreateWorkGroupOutput, error) {
+	if c.workgroup != "" {
+		return nil, &athenatypes.InvalidRequestException{Message: aws.String("workgroup is already created")}
+	}
+
+	c.workgroup = *params.Name
+
+	return &athena.CreateWorkGroupOutput{}, nil
+}
+
+func (c *fakeBootstrapInfraClient) UpdateTable(ctx context.Context, params *glue.UpdateTableInput, optFns ...func(*glue.Options)) (*glue.UpdateTableOutput, error) {
+	if c.table == "" {
+		return nil, &gluetypes.InvalidInputException{Message: aws.String("the table does not exist")}
+	}
+
+	return &glue.UpdateTableOutput{}, nil
+}
+
+func (c *fakeBootstrapInfraClient) CreateTable(ctx context.Context, params *glue.CreateTableInput, optFns ...func(*glue.Options)) (*glue.CreateTableOutput, error) {
+	if c.table != "" {
+		return nil, &gluetypes.AlreadyExistsException{Message: aws.String("table already exists")}
+	}
+
+	c.table = *params.TableInput.Name
+
+	return &glue.CreateTableOutput{}, nil
+}
+
+// Creates a new database in a Data Catalog.
+func (c *fakeBootstrapInfraClient) CreateDatabase(ctx context.Context, params *glue.CreateDatabaseInput, optFns ...func(*glue.Options)) (*glue.CreateDatabaseOutput, error) {
+	if c.database != "" {
+		return nil, &gluetypes.AlreadyExistsException{Message: aws.String("database already exists")}
+	}
+
+	c.database = *params.DatabaseInput.Name
+
+	return &glue.CreateDatabaseOutput{}, nil
+}
+
+func TestValidateAndParseS3Input(t *testing.T) {
+	tt := []struct {
+		desc        string
+		eca         *ecatypes.ExternalCloudAuditSpec
+		errExpected bool
+	}{
+		{
+			desc:        "nil external cloud audit spec",
+			errExpected: true,
+		},
+		{
+			desc:        "unique s3 uris for all three buckets",
+			errExpected: true,
+			eca: &ecatypes.ExternalCloudAuditSpec{
+				SessionsRecordingsURI:  "s3://sessions/prefix",
+				AuditEventsLongTermURI: "s3://events/prefix",
+				AthenaResultsURI:       "s3://results/query_results",
+			},
+		},
+		{
+			desc:        "valid s3 configuration",
+			errExpected: false,
+			eca: &ecatypes.ExternalCloudAuditSpec{
+				SessionsRecordingsURI:  "s3://lts/prefix",
+				AuditEventsLongTermURI: "s3://lts/prefix",
+				AthenaResultsURI:       "s3://results/query_results",
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.desc, func(t *testing.T) {
+			ltsBucket, transientBucket, err := validateAndParseS3Input(tc.eca)
```

Comment:
Could we test this via our public APIs?

Reply:
Added additional tests in [ffc9570](https://github.com/gravitational/teleport/pull/33272/commits/ffc95703e2d1522c1ebfb67a951be6683b284046) that help test the private `validateAndParseS3Input` method.